cmd/k-license/k-license.go

```diff
@@ -0,0 +1,232 @@
+	fileExtension    string // store file extension strings like ".sh", "Makefile", etc.
+	templateFileName string // store template file names like "boilerplate.sh.txt", etc
+}
+
+var templateFileTypes = []templateFileType{
+	{".sh", "boilerplate.sh.txt"},
+	{"Makefile", "boilerplate.Makefile.txt"},
+	{"Dockerfile", "boilerplate.Dockerfile.txt"},
+	{".py", "boilerplate.py.txt"},
+	{".go", "boilerplate.go.txt"},
+}
+
+var opts = &Options{}
+
+func main() {
+	rootCmd := &cobra.Command{
+		Use:   "k-license",
+		Short: "Tool for Adding license Headers",
+	}
+	addCmd := &cobra.Command{
+		Use:   "add",
+		Short: "Add Headers to files",
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			return opts.Run()
+		},
+	}
+
+	addCmd.Flags().StringVar(&opts.templatesDir, "templates", "../../hack/boilerplate", "directory containing license templates")
+	addCmd.Flags().StringSliceVarP(&opts.excludeDirs, "exclude", "e", excludeDirsLocations, "comma-separated list of directories to exclude")
+	addCmd.Flags().StringVar(&opts.path, "path", ".", "Defaults to Current directory")
+	addCmd.Flags().BoolVar(&opts.confirm, "confirm", false, "")
+	rootCmd.AddCommand(addCmd)
+	if err := rootCmd.Execute(); err != nil {
+		os.Exit(1)
+	}
+
+}
+func (opts *Options) Run() error {
+	//excludeList := strings.Split(opts.excludeDirs, ",")
+	files := 0
+	err := filepath.WalkDir(opts.path, func(path string, info fs.DirEntry, err error) error {
+		if info.IsDir() && contains(opts.excludeDirs, info.Name()) {
+			return filepath.SkipDir
+		}
+		if !info.IsDir() && (isCodeFile(path) || isBuildFile(path)) && (!hasLicense(path) && !isGenerateFile(path)) {
+			currentYear := strconv.Itoa(time.Now().Year())
+			if opts.confirm {
+				err := addLicense(path, opts.templatesDir, currentYear)
+				if err != nil {
+					return err
+				}
+				fmt.Printf("Modified %s file\n", path)
+			}
+			files++
+		}
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	if opts.confirm {
+		fmt.Printf("Modified %v files\n", files)
+	} else {
+		fmt.Printf("Skipping modification. Dry running \t")
+		fmt.Printf("%v files will be modified\n", files)
+	}
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+	return nil
+}
+func contains(list []string, str string) bool {
+	for _, item := range list {
+		if item == str {
+			return true
+		}
+	}
+	return false
+}
+
+func isCodeFile(path string) bool {
+	return codeFileExts[strings.ToLower(filepath.Ext(path))]
+}
+
+func isBuildFile(path string) bool {
+	return buildFileExts[filepath.Base(path)]
+}
+
+func isGenerateFile(path string) bool {
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return false
+	}
+	for _, ft := range GENERATED_GO_MARKERS {
+		if strings.Contains(string(data), ft) {
+			fmt.Printf("Skipping File: %s since this is autogenerated file \n", path)
+			return true
+		}
+	}
+	return false
+}
+
+func hasLicense(path string) bool {
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return false
```

Comment:
@mohitsharma-in, answering below for https://github.com/kubernetes/org/pull/4077#discussion_r1165100234

Currently the function is returning a `false` boolean value in case if  ` err != nil`.  Returning the error message along with boolean, would be helpful for debugging.